lib/crawler/rule_engine/base.rb

```diff
@@ -69,7 +69,7 @@ def output_crawl_result_outcome(crawl_result) # rubocop:disable Metrics/AbcSize,
         end
 
         return denied_outcome(:content_type_denied, crawl_result.error) if crawl_result.unsupported_content_type?
-        return denied_outcome(:fatal_error_denied, crawl_result.error) if crawl_result.fatal_error?
+        return denied_outcome(:error_denied, crawl_result.error) if crawl_result.error?
```

Comment:
What's the difference here (fatal_error vs error)?

Reply:
Fatal error seemed to be for internal or spec use but I couldn't find any actual usage of it: https://github.com/elastic/crawler/blob/1c92bb5fb79463a33e769a8a3520732d22cd3230/lib/crawler/data/crawl_result/error.rb#L16

After confirming that I've also removed the const and all other references to `fatal_error`.

Reply:
Well, I was completely wrong and removing this toppled the entire house-of-cards 😅 

The fatal error code is used as a default when creating errors when the error wasn't an HTTP error (like 4xx status codes). So it's used to represent things like timeouts or other internal errors.
Before, crawler would create a document even if the status code from the server was an error. Now, both "fatal" errors and regular errors are handled identically, so in both cases crawler won't create an ES document.

Reply:
I'm going to merge this and do a quick follow-up PR that renames fatal_error to internal_error and adds some comments